frontend/app/views/spree/orders/edit.html.erb

```diff
@@ -36,7 +36,11 @@
             </div>
 
             <% if Spree::Frontend::Config[:coupon_codes_enabled] %>
-              <%= render 'coupon_code', promotion: @order.promotions.coupons.first, order_form: order_form, classes: '' %>
+              <%= render 'coupon_code',
+                promotion: @order.valid_promotions.first&.promotion,
```

Comment:
@KirillKayumov `valid_promotions` will return all promotions applicated for this Order or Order Line Items, including non-coupon code promotions like free shipping etc so we need to fetch only valid coupon code promotion, maybe a new method in Order model?